scripts/vpn-killswitch.sh

```diff
@@ -106,37 +108,49 @@ get_wan_interface() {
 }
 
 firewall_rules() {
-    [ -z "$BRIDGE_INTERFACE" ] && { logger -st "$SCRIPT_TAG" "Bridge interface is not set"; exit 1; }
+    [ -z "$FOR_BRIDGE_INTERFACES" ] && { logger -st "$SCRIPT_TAG" "Bridge interfaces is not set"; exit 1; }
+    if echo "$FOR_BRIDGE_INTERFACES" | grep -q "br+"; then
+        logger -st "$SCRIPT_TAG" "Applying firewall rules to all interfaces [br+]"
+        FOR_BRIDGE_INTERFACES="br+" # sanity set, just in case one sets "br0 br+ br23"
+    else
+        for _BRIDGE_INTERFACE in $FOR_BRIDGE_INTERFACES; do
+            if ! nvram show 2>/dev/null | grep -q -E "ifname=$_BRIDGE_INTERFACE$"; then
```

Comment:
Wouldn't it be better to check for physical interface existence in the system here?

For example like this:
```bash
IF=br0
if ip link show | grep ": $IF" | grep -q "mtu"; then
     echo "$IF exists!"
fi
```

Someone might want to use this script with their "exotic" configuration where their interface might not even be mentioned in nvram.

Or this whole error check block could be completely removed if iptables ignores non-existing interfaces (if it errors out this block shall be kept).

Reply:
Absolutely, your proposal to use `ip link` over `nvram` is more reliable.
As for the check block, iptables indeed ignores non-existing interfaces when applying rules, but I added it for the following two reasons:
1) To make sure inputs are being checked before applying the rules
2) If changes are made later via the asus web interface, we'll see it error out in the router logs and remember we had that rule set that is no longer relevant or needs to be updated

We could technically also add it as a warning instead of error considering it is informative log:
```diff
for _BRIDGE_INTERFACE in $BRIDGE_INTERFACES; do
-     if ! nvram show 2>/dev/null | grep -q -E "ifname=$_BRIDGE_INTERFACE$"; then
-        logger -st "$SCRIPT_TAG" "No interfaces matching [$_BRIDGE_INTERFACE] bridge interface found"
-        exit 1
+    if ! ip link show | grep ": $_BRIDGE_INTERFACE" | grep -q "mtu"; then
+        logger -st "$SCRIPT_TAG" "Warning: Couldn't find matching bridge interface for [$_BRIDGE_INTERFACE]"
+        exit 0
    fi
done
```
I would suggest against adding the rule if the provided bridge interface does not exists, since let's remember that this runs every minute. wdyt?